src/tabmat/constructor.py

```diff
@@ -220,99 +255,83 @@ def from_polars(
-                cat_missing_name=cat_missing_name,
-            )
-            if len(cat.categories) < cat_threshold:
-                (
-                    X_dense_F,
-                    X_sparse,
-                    dense_idx,
-                    sparse_idx,
-                ) = _split_sparse_and_dense_parts(
-                    sps.csc_matrix(cat.tocsr(), dtype=dtype),
-                    threshold=sparse_threshold,
-                    column_names=cat.get_names("column"),
-                    term_names=cat.get_names("term"),
-                )
-                matrices.append(X_dense_F)
-                is_cat.append(True)
-                matrices.append(X_sparse)
-                is_cat.append(True)
-                if cat_position == "expand":
-                    indices.append(mxcolidx + dense_idx)
-                    indices.append(mxcolidx + sparse_idx)
-                    mxcolidx += len(dense_idx) + len(sparse_idx)
-                elif cat_position == "end":
-                    indices.append(dense_idx)
-                    indices.append(sparse_idx)
-
-            else:
-                matrices.append(cat)
-                is_cat.append(True)
-                if cat_position == "expand":
-                    indices.append(mxcolidx + np.arange(cat.shape[1]))
-                    mxcolidx += cat.shape[1]
-                elif cat_position == "end":
-                    indices.append(np.arange(cat.shape[1]))
-        elif coldata.dtype.is_numeric():
-            if (coldata != 0).mean() <= sparse_threshold:
-                sparse_columns.append(coldata.name)
-                sparse_indices.append(mxcolidx)
-                mxcolidx += 1
-            else:
-                dense_columns.append(coldata.name)
-                dense_indices.append(mxcolidx)
-                mxcolidx += 1
+    return _from_dataframe(
+        df,
+        engine="pandas",
+        dtype=dtype,
+        sparse_threshold=sparse_threshold,
+        cat_threshold=cat_threshold,
+        object_as_cat=object_as_cat,
+        cat_position=cat_position,
+        drop_first=drop_first,
+        categorical_format=categorical_format,
+        cat_missing_method=cat_missing_method,
+        cat_missing_name=cat_missing_name,
+    )
 
-        else:
-            ignored_cols.append(coldata.name)
 
-    if len(ignored_cols) > 0:
-        warnings.warn(
-            f"Columns {ignored_cols} were ignored. Make sure they have a valid dtype."
-        )
-    if dense_columns:
-        matrices.append(_dense_matrix(df, dense_columns, dtype))
-        indices.append(dense_indices)
-        is_cat.append(False)
-    if sparse_columns:
-        matrices.append(_sparse_matrix(df, sparse_columns, dtype))
-        indices.append(sparse_indices)
-        is_cat.append(False)
+def from_polars(
+    df,
+    dtype: np.dtype = np.float64,
+    sparse_threshold: float = 0.1,
+    cat_threshold: int = 4,
+    object_as_cat: bool = False,
```

Comment:
This argument is not documented.

It's also a bit misleading in that we only convert string columns (rather than object columns).